### PR TITLE
Fix/1990/confirmation on taking over evaluation

### PR DESCRIPTION
**Pull request by aquibmoin786**
Targets #1990

### Description
- We are add evaluate button. When click the evaluate button then open the popup model. In this popup model have two button one is request  and another is cancel. When clicking the request button then send a mail request to existing assignee for Application Handover request link with Application Id. Upon clicking this link, the existing assignee will be redirected to the portal. Show on portal Application is assignee to requested person.

### Checklist:
- [x] I have performed a self-review of my own code.

### CODE REVIEW

**Comment by gautam071** [comment]
@aquibmoin786 Fix the Checks

**Review by P4NK4J — CHANGES_REQUESTED** [review]
@aquibmoin786 If an application is already assigned to a person clicking on evaluate button, the popup should not appear, it should only as for handover if the application is assigned to other person than the logged in user. 
Also, in the email, there are a couple of things to be changed, 
1. Hey Scheduled person is to be replaced by the person name to whom the application is assigned
2. Provide application details, in the email, The |*APPLICATION TYPE*| application of |*APPLICANT NAME*| applied for |*JOB ROLE*| is requested for evaluation by |*PERSON SENDING THE REQUEST*|. Please click the link to handover the application.
|APPLICATION TYPE| can be job or internship, data will be available in applications table
3. Also, correct the spelling mistakes in the email template

File: Modules/HR/Http/Controllers/Recruitment/ApplicationController.php [hunk 1]
```diff
@@ -260,6 +261,11 @@ public function viewOfferLetter(Application $application)
 
     public function request(Application $application)
     {
+        $currentUser = Auth::user()->id;
+        $currentAssigneePerson = $application->latestApplicationRound->scheduledPerson->id;
+        if($currentAssigneePerson === $currentUser) {
+            return redirect()->back()->with('status', "You are not assign to this application So you cann't take this action");
```
Comment (gautam071) [inline Modules/HR/Http/Controllers/Recruitment/ApplicationController.php:267]:
```suggestion
            return redirect()->back()->with('status', "You are not assign to this application So you can't take this action");
```

File: Modules/HR/Resources/views/application/application-handover-request.blade.php [hunk 2]
```diff
@@ -1,3 +1,7 @@
-<p>Hey Scheduled person, <br> Please click the link and handover the application id : {{$application->id}}</p>
+@php
+	$assignee = $application->latestApplicationRound->scheduledPerson;
+@endphp
+{{-- $jobType = $this->getApplicationType(); --}}
```
Comment (gautam071) [inline Modules/HR/Resources/views/application/application-handover-request.blade.php:4]: Remove the unwanted code

**Review by gautam071 — CHANGES_REQUESTED** [review]

File: .husky/_/husky.sh [hunk 3]
```diff
@@ -1,7 +1,9 @@
 #!/bin/sh
 if [ -z "$husky_skip_init" ]; then
   debug () {
-    [ "$HUSKY_DEBUG" = "1" ] && echo "husky (debug) - $1"
+    if [ "$HUSKY_DEBUG" = "1" ]; then
+      echo "husky (debug) - $1"
+    fi
```
Comment (P4NK4J) [inline .husky/_/husky.sh:6]: Why we need these changes in this PR ?

File: Modules/HR/Http/Controllers/Recruitment/ApplicationController.php [hunk 4]
```diff
@@ -257,4 +257,24 @@ public function viewOfferLetter(Application $application)
             'content-type' => 'application/pdf',
         ]);
     }
+
+    public function request(Application $application)
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/ApplicationController.php:261]: What's the role of this function ? Can we rename it to something meaningful from readability perspective ? Maybe `applicationHandoverRequest` ? Let me know your thoughts

File: Modules/HR/Http/Controllers/Recruitment/ApplicationController.php [hunk 5]
```diff
@@ -257,4 +257,24 @@ public function viewOfferLetter(Application $application)
...
+    {
+        $currentAssignee = $application->latestApplicationRound->scheduledPerson->email;
+        Mail::to($currentAssignee)->send(new ApplicationHandover($application));
+
+        return redirect()->back()->with('status', 'You email has successfully been sent');
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/ApplicationController.php:266]:
```suggestion
        return redirect()->back()->with('status', 'Your request has successfully been sent');
```

File: Modules/HR/Http/Controllers/Recruitment/ApplicationController.php [hunk 6]
```diff
@@ -257,4 +257,24 @@ public function viewOfferLetter(Application $application)
...
+    }
+
+    public function acceptHandoverRequest(Application $application, User $user)
+    {
+        $applicationRound = $application->latestApplicationRound;
+        $applicationRound->update([
+            'scheduled_person_id' => $user->id
+        ]);
+
+        $status = 'Successful Assign to ' . $user->name;
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/ApplicationController.php:276]:
```suggestion
        $status = 'Successful Assigned to ' . $user->name;
```

File: Modules/HR/Resources/views/application/application-handover-request.blade.php [hunk 7]
```diff
@@ -0,0 +1,6 @@
+@php
+	$assignee = $application->latestApplicationRound->scheduledPerson;
+@endphp
+<p>Hey {{$assignee->name}}, <br> <br>The {{$application->job->type}} application of {{$application->applicant->name}} applied for {{$application->job->title}} is requested for evaluation by {{auth()->user()->name}}. Please click the link to handover the application : {{$application->name}}</p>
```
Comment (P4NK4J) [inline Modules/HR/Resources/views/application/application-handover-request.blade.php:4]: Why do we need $application->name here ? Do we have any such attribute in the database ?

File: Modules/HR/Resources/views/application/application-handover-request.blade.php [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+<a href="{{ route('application.handover.confirmation', ['application' => $application->id, 'user' => auth()->user()->id])}}">Handover application</a>
+<p>Thanks,<br>HR,<br>Coloredcow</p>
```
Comment (P4NK4J) [inline Modules/HR/Resources/views/application/application-handover-request.blade.php:6]: Please correct HTML formatting for this template.

File: Modules/HR/Resources/views/application/render-application-row.blade.php [hunk 9]
```diff
@@ -30,11 +30,39 @@
...
+								    <p class="text-secondary">This application is already assigned to {{$assignee->name}}, to evaluate this, a confirmation would be needed from their end. Please click the request button to request the handover.</p>
+							    </div>
+							    <div class="modal-footer justify-content-between">
+								    <a href="{{ route('application.handover', $application) }}" class="btn btn-primary">Request</a></button>
+								    <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+							    </div>
+						    </div>
+					    </div>
+				    </div>
+				@endif
```
Comment (P4NK4J) [inline Modules/HR/Resources/views/application/render-application-row.blade.php:63]: Formatting seems off, please check

File: Modules/HR/Routes/web.php [hunk 10]
```diff
@@ -44,6 +44,8 @@
             ->names(['index' => 'hr.tags.index', 'edit' => 'hr.tags.edit', 'update' => 'hr.tags.update', 'store' => 'hr.tags.store', 'destroy' => 'hr.tags.delete']);
 
         Route::prefix('recruitment')->namespace('Recruitment')->group(function () {
+            Route::get('application/{application}/handover', 'JobApplicationController@request')->name('application.handover');
+            Route::get('application/{application}/assign-to/{user}', 'JobApplicationController@acceptHandoverRequest')->name('application.handover.confirmation');
```
Comment (P4NK4J) [inline Modules/HR/Routes/web.php:48]: What we are passing in {user} here ?

File: Modules/HR/Routes/web.php [hunk 11]
```diff
@@ -15,7 +15,7 @@
         Route::get('/hiring', 'HiringController@index')->name('hr-new.hiring');
         Route::get('/evaluation/segments', 'EvaluationController@index')->name('hr.evaluation');
         Route::get('evaluation/segment/{segmentID}/parameters', 'EvaluationController@segmentParameters')->name('hr.evaluation.segment-parameters');
-        // Route::get('evaluation/segment/{segmentID}/parameters', 'EvaluationController@segmentParameters')->name('hr.evaluation.segment-parameters');
+        Route::get('evaluation/segment/{segmentID}/parameters', 'EvaluationController@segmentParameters')->name('hr.evaluation.segment-parameters');
```
Comment (P4NK4J) [inline Modules/HR/Routes/web.php:18]: Why we need to change this ?

File: Modules/HR/Resources/views/application/render-application-row.blade.php [hunk 12]
```diff
@@ -86,10 +114,7 @@ class="mr-1 text-decoration-none">
 		</div>
 	</td>
 	<td class="">
-		@php
-		$assignee = $application->latestApplicationRound->scheduledPerson;
-		@endphp
-		<img src="{{$assignee->avatar}}" alt="{{$assignee->name}}" class="w-25 h-25 rounded-circle"
```
Comment (P4NK4J) [inline Modules/HR/Resources/views/application/render-application-row.blade.php:92]: how you are getting $assignee, after removing these lines ?

File: routes/web.php [hunk 13]
```diff
@@ -93,7 +93,7 @@
 
     Route::post('projects/{project}/add-employee', 'ProjectController@addEmployee');
     Route::post('projects/{project}/remove-employee', 'ProjectController@removeEmployee');
-    Route::get('my-projects/{employee}', 'hr\Employees\EmployeeController@showProjects')->name('projects.my-projects');
+    // Route::get('my-projects/{employee}', 'hr\Employees\EmployeeController@showProjects')->name('projects.my-projects');
```
Comment (P4NK4J) [inline routes/web.php:96]: Why you have commented out this line ?

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Left some comments, please check.

File: .husky/_/husky.sh (hunk 3 above)
Comment (aquibmoin786) [reply .husky/_/husky.sh:6]: It's auto generated.

File: Modules/HR/Http/Controllers/Recruitment/ApplicationController.php (hunk 4 above)
Comment (aquibmoin786) [reply Modules/HR/Http/Controllers/Recruitment/ApplicationController.php:261]: Changes

File: Modules/HR/Http/Controllers/Recruitment/ApplicationController.php (hunk 5 above)
Comment (aquibmoin786) [reply Modules/HR/Http/Controllers/Recruitment/ApplicationController.php:266]: Changes

File: Modules/HR/Http/Controllers/Recruitment/ApplicationController.php (hunk 6 above)
Comment (aquibmoin786) [reply Modules/HR/Http/Controllers/Recruitment/ApplicationController.php:276]: Changes

File: Modules/HR/Resources/views/application/application-handover-request.blade.php (hunk 7 above)
Comment (aquibmoin786) [reply Modules/HR/Resources/views/application/application-handover-request.blade.php:4]: By mistake added. I have remove that.

File: Modules/HR/Resources/views/application/render-application-row.blade.php (hunk 12 above)
Comment (aquibmoin786) [reply Modules/HR/Resources/views/application/render-application-row.blade.php:92]: I have declare $assignee globally and for use this line `<p class="text-secondary">This application is already assigned to {{$assignee->name}}, to evaluate this, a confirmation would be needed from their end. Please click the request button to request the handover.</p>` and below line  ```
avatar}}" alt="{{$assignee->name}}" class="w-25 h-25 rounded-circle"
			data-toggle="tooltip" data-placement="top" title="{{$assignee->name}}">
```

File: Modules/HR/Routes/web.php (hunk 11 above)
Comment (aquibmoin786) [reply Modules/HR/Routes/web.php:18]: I have changes it

File: Modules/HR/Routes/web.php (hunk 10 above)
Comment (aquibmoin786) [reply Modules/HR/Routes/web.php:48]: I use for getting user id and user name

File: routes/web.php (hunk 13 above)
Comment (aquibmoin786) [reply routes/web.php:96]: I have changes it.

File: .husky/_/husky.sh (hunk 3 above)
Comment (P4NK4J) [reply .husky/_/husky.sh:6]: Can we remove it from this PR ?

File: Modules/HR/Resources/views/application/application-handover-request.blade.php [hunk 14]
```diff
@@ -0,0 +1,10 @@
+@php
+    $assignee = $application->latestApplicationRound->scheduledPerson;
+@endphp
+<p>
```
Comment (P4NK4J) [inline Modules/HR/Resources/views/application/application-handover-request.blade.php:4]: Let's keep it as it was in the render blade and use the $assignee variable here ?

File: Modules/HR/Resources/views/application/render-application-row.blade.php [hunk 15]
```diff
@@ -7,120 +7,145 @@
...
+                                <div class="modal-footer justify-content-between">
+                                    <a href="{{ route('application.handover', $application) }}" class="btn btn-primary">Request</a></button>
+                                    <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                @endif
+            </div>
+
```
Comment (P4NK4J) [inline Modules/HR/Resources/views/application/render-application-row.blade.php:65]: Remove this space

File: routes/web.php (hunk 13 above)
Comment (P4NK4J) [reply routes/web.php:96]: Its still showing commented here, am I missing something ?

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Left some comments, please check

File: Modules/HR/Http/Controllers/Recruitment/ApplicationController.php [hunk 16]
```diff
@@ -21,6 +21,7 @@
 use Modules\HR\Http\Requests\Recruitment\CustomApplicationMailRequest;
 use Modules\HR\Services\ApplicationService;
 use Modules\User\Entities\User;
+use App\Mail\ApplicationHandover;
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/ApplicationController.php:24]: Let's put it inside HR module Email folder

File: app/Mail/ApplicationHandover.php [hunk 17]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Mail;
```
Comment (P4NK4J) [inline app/Mail/ApplicationHandover.php:3]: Let's put this inside HR email folder

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Minor comment, else looks good

**Review by P4NK4J — COMMENTED** [review]
@aquibmoin786 Let's keep the routes file changes as they were in the master branch. Please make these changes.  in HR routes and main web.php

**Review by P4NK4J — APPROVED** [review]
Looks good.

File: Modules/HR/Resources/views/application/render-application-row.blade.php [hunk 18]
```diff
@@ -99,7 +123,7 @@ class="mr-1 text-decoration-none">
 			class="{{ config("constants.hr.status.$application->status.class") }} badge-pill mr-1 mb-1 fz-12">{{ config("constants.hr.status.$application->status.title") }}</span>
 		@endif
 		@if (!$application->latestApplicationRound->scheduled_date &&
-		$application->latestApplicationRound->round->name != "Telephonic Interview" && $application->latestApplicationRound->round->name != "Team Interaction Round")
+		$application->latestApplicationRound->hr_round_id != 14)
```
Comment (gautam071) [inline Modules/HR/Resources/views/application/render-application-row.blade.php:126]: Don't hard code the id

File: Modules/HR/Resources/views/application/render-application-row.blade.php [hunk 19]
```diff
@@ -129,8 +153,8 @@ class="{{ config("constants.hr.status.$application->status.class") }} badge-pill
...
-		</span>
-		@endforeach
-		</span>
-	</td>
-</tr>
+        </span>
+        @endforeach
+        </span>
+    </td>
+</tr>
```
Comment (gautam071) [inline Modules/HR/Resources/views/application/render-application-row.blade.php:160]: Leave a line at the end

File: resources/views/mail/hr/application-handover-request-email.blade.php [hunk 20]
```diff
@@ -0,0 +1,7 @@
+<p>
+    Hey {{$application->latestApplicationRound->scheduledPerson->name}}, <br><br>The {{$application->job->type}} application of {{$application->applicant->name}} applied for {{$application->job->title}} is requested for evaluation by {{auth()->user()->name}}. Please click the link to handover the application :
+</p>
+    <a href="{{ route('application.handover.confirmation', ['application' => $application->id, 'user' => auth()->user()->id])}}">Handover application</a>
+<p>
+    Thanks,<br>HR,<br>Coloredcow
+</p>
```
Comment (gautam071) [inline resources/views/mail/hr/application-handover-request-email.blade.php:7]: leave a line at the end

**Review by gautam071 — COMMENTED** [review]
Left some comments

**Review by gautam071 — APPROVED** [review]
Looks good

**Review by P4NK4J — APPROVED** [review]
Looks good.

**Pull request merged**